tools/c7n_index/c7n_index/metrics.py

```diff
@@ -309,16 +313,33 @@ def index_account_resources(config, account, region, policy, date):
     key_prefix = "accounts/{}/{}/policies/{}".format(
         account['name'], region, policy['name'])
 
+    # Look for AWS profile in config before Instance role
     records = s3_resource_parser.record_set(
-        lambda : assumed_session(account['role'], 'PolicyIndex'),
-        bucket,
-        key_prefix,
-        date,
-        specify_hour=True)
+    lambda: SessionFactory(region, profile=account.get('profile'),
+        assume_role=account.get('role'))(),
+    bucket,
+    key_prefix,
+    date,
+    specify_hour=True)
+
 
     for r in records:
+        # Adding Custodian vars to each record
         r['c7n:MatchedPolicy'] = policy['name']
-    
+        r['c7n:AccountNumber'] = account['id']
+
+        # Expand tags
+        # Tags are stored in the following format:
+        # Tags: [ {'key': 'mykey', 'val': 'myval'}, {'key': 'mykey2', 'val': 'myval2'} ]
+        # and this makes searching for tags difficult. We will convert them to:
+        # Tags: ['mykey': 'myval', 'mykey2': 'myval2']
+        if r.get('Tags'):
+            tags = {}
```

Comment:
one liner for the same = r['Tags'] = {t['Key']: t['Value'] for t in r['Tags']

minor consideration for putting this in a separate field since its not a true representation of the aws resource (has a delta), but given its not system of record and we can reindex it seems okay.